modules/db/src/main/dsl.scala

```diff
@@ -104,19 +104,16 @@ trait dsl {
     $doc("$setOnInsert" -> $doc((Seq(item) ++ items): _*))
   }
 
-  def $set(item: ElementProducer, items: ElementProducer*): Bdoc = {
-    $doc("$set" -> $doc((Seq(item) ++ items): _*))
-  }
+  def $set(items: ElementProducer*): Bdoc =
+    $doc("$set" -> items.nonEmpty ?? $doc(items: _*))
+
+  def $unset(fields: Iterable[String]): Bdoc =
+    $doc("$unset" -> fields.nonEmpty ?? $doc(fields.map(k => (k, BSONString("")))))
```

Comment:
if I understand correctly, before:
```scala
$unset(Nil) // $doc()
```
and after:
```scala
$unset(Nil) // $doc("$unset" -> $doc())
```
why is it better?

Reply:
`coll.findAndUpdate(..., update = $unset(Nil))`

Reply:
```
lichess> db.test.findOneAndUpdate({},{})
MongoInvalidArgumentError: Update document requires atomic operators
lichess> db.test.findOneAndUpdate({},{$unset:{}})
{ _id: 1 }
```
I see now